src/main/java/org/apache/maven/model/jdom/util/JDomCleanupHelper.java

```diff
@@ -0,0 +1,205 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.jdom2.Comment;
+import org.jdom2.Content;
+import org.jdom2.Element;
+import org.jdom2.filter.ElementFilter;
+import org.jdom2.util.IteratorIterable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_PROFILE;
+import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_PROFILES;
+
+/**
+ * JDom cleanup methods for:
+ * <ol>
+ * <li>profiles</li>
+ * </ol>
+ *
+ * @author https://github.com/eva-mueller-coremedia  (for <a href="https://github.com/CoreMedia/maven-jdom-parser">Maven JDom Parser</a>, version 3.0)
+ */
+public class JDomCleanupHelper {
+
+  /**
+   * Remove all empty profiles and profile tags.<br>
+   * Empty is defined as
+   * <ul>
+   * <li>for tag <b>profiles</b>: there are no profile tags</li>
+   * <li>for tag <b>profile</b>: Either there are only the tags activation and id present or all other tags are empty</li>
+   * </ul>
+   * Thus, empty tags may contain comments which will be removed as well.
+   *
+   * @param rootElement the root element.
+   */
+  public static void cleanupEmptyProfiles(Element rootElement) {
+    IteratorIterable<Element> filteredElements = rootElement.getDescendants(new ElementFilter(POM_ELEMENT_PROFILES));
+    List<Element> profiles = new ArrayList<>();
+    for (Element profilesElement : filteredElements) {
+      profiles.add(profilesElement);
+    }
+    for (Element profilesElement : profiles) {
+      removeElementWithEmptyChildren(profilesElement,
+              POM_ELEMENT_PROFILE,
+              Arrays.asList(JDomCfg.POM_ELEMENT_ID, JDomCfg.POM_ELEMENT_ACTIVATION)
+      );
+      if (!profilesElement.getDescendants(new ElementFilter(POM_ELEMENT_PROFILE)).hasNext()) {
+        JDomUtils.removeChildAndItsCommentFromContent(profilesElement.getParentElement(), profilesElement);
+      }
+    }
+  }
+
+  /**
+   * Remove empty properties tags with special parents.<br>
+   * Empty tags may contain comments which will be removed as well.
+   *
+   * @param rootElement the root element.
+   * @param parentTags  Allowed parent tags.
+   */
+  public static void cleanupEmptyProperties(Element rootElement, List<String> parentTags) {
```

Comment:
I wouldn't delete empty properties, but when deleting a property its comments should be deleted, too.